pkg/apis/extensions/types.go

```diff
@@ -208,7 +208,7 @@ type DeploymentSpec struct {
 	Selector map[string]string `json:"selector,omitempty"`
 
 	// Template describes the pods that will be created.
-	Template *api.PodTemplateSpec `json:"template,omitempty"`
+	Template api.PodTemplateSpec `json:"template"`
```

Comment:
If we were to add TemplateRef, it would be in a new apiVersion, anyway. Making template a required field for now is more understandable for users.

TemplateRef is one of several issues to consider in the controller API overhaul: #14961. I don't know when I'll have time to get to it, though.